### PR TITLE
utils/bind-data-test-attributes: Handle failing `set()`  with assert instead of warning 

### PR DESCRIPTION
**Pull request by mathieupoteriepeopledoc**
Wrote an integration test in order to test that the error is thrown when a data-test attribute is set on a tagless component.

close #205

### CODE REVIEW

**Comment by Turbo87** [comment]
looks like CI is failing

**Comment by mathieupoteriepeopledoc** [comment]
@Turbo87 I will take a look this afternoon. It was working locally. :)

**Comment by mathieupoteriepeopledoc** [comment]
I updated the PR, in order to pass the test for every ember versions. I keep working on it and will let you know when I found a satisfying solution.

**Comment by mathieupoteriepeopledoc** [comment]
@Turbo87 , the CI is failing, but on the same stage than the other PR (in the ember-release scenario). A colleague and I try to investigate on this issue. If we find anything, we will let you know.

**Comment by Turbo87** [comment]
@mathieupoteriepeopledoc CI should be fixed now. can you rebase your changes on top of `master`?

**Comment by mathieupoteriepeopledoc** [comment]
@Turbo87 , this is done, thanks to @xcambar  :)

File: tests/unit/utils/bind-data-test-attributes-test.js [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+import config from 'dummy/config/environment';
 
 import bindDataTestAttributes from 'ember-test-selectors/utils/bind-data-test-attributes';
 
 module('Unit | Utility | bind data test attributes');
 
 test('it adds missing attributeBindings array', function(assert) {
   let Fixture = EmberObject.extend({
-    'data-test-from-factory': 'foo',
+    'data-test-from-factory': 'foo'
```
Comment (Turbo87) [inline tests/unit/utils/bind-data-test-attributes-test.js]: can you remove these unrelated changes? the PR should only contain the changed functionality. Styling changes should be proposed in a seperate PR.

**Comment by mathieupoteriepeopledoc** [comment]
@Turbo87, this is fixed, and git history of this PR has been cleaned.

File: tests/unit/utils/bind-data-test-attributes-test.js [hunk 2]
```diff
@@ -119,18 +120,29 @@ test('it skips if attributeBindings is a computed property', function(assert) {
   assert.deepEqual(instance.get('attributeBindings'), ['foo']);
 });
 
-test('it skips if tagName is empty', function(assert) {
-  let Fixture = EmberObject.extend({
-    tagName: '',
-    'data-test-from-factory': 'foo',
+if (!config.stripTestSelectors) {
```
Comment (Turbo87) [inline tests/unit/utils/bind-data-test-attributes-test.js]: I don't understand this. Why are you checking `config.stripTestSelectors` here? The `bindDataTestAttributes()` function doesn't seem to care about that config option 🤔

**Comment by mathieupoteriepeopledoc** [comment]
@Turbo87 , sorry it was some code remaining from an integration test made previously. 
I think it's okay now, sorry for the many feedbacks ;)
